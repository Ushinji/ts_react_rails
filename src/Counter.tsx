import * as React from 'react';

const Counter: React.FC = () => {
  const [count, SetCount] = React.useState(0);

  const handleClick = () => {
    SetCount(count + 1);
  };

  return (
    <div>
      <h1>カウンター</h1>
      <button type="button" onClick={handleClick}>
        ＋
      </button>
      <div>{count}</div>
    </div>
  );
};

export default Counter;
