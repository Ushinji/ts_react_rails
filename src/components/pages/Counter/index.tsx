import * as React from 'react';
import styled from 'styled-components';
import Button from '@/components/atoms/Button';
import useBool from '@/modules/useBool';

const Container = styled.div`
  background-color: white;
  box-shadow: 0px 0px 4px rgba(51, 23, 20, 0.2);
  padding: 16px;
  width: 340px;
  border-radius: 4px;
`;

const Title = styled.div`
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 16px;
`;

const Counter: React.FC = () => {
  const [submitting, startSubmit, finishSubmit] = useBool(false);
  const [count, SetCount] = React.useState(0);

  const handleClick = () => {
    startSubmit();
    setTimeout(() => {
      SetCount(count + 1);
      finishSubmit();
    }, 1000);
  };

  return (
    <Container>
      <Title>
        カウンター：
        {count}
      </Title>
      <Button type="button" onClick={handleClick} submitting={submitting}>
        Count up
      </Button>
    </Container>
  );
};

export default Counter;
