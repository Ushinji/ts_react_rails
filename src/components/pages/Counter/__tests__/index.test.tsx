import * as React from 'react';
import { render } from 'react-testing-library';
import Counter from '..';

describe('pages/Counter', () => {
  it('Snapshot test', () => {
    const { asFragment } = render(<Counter />);
    expect(asFragment()).toMatchSnapshot();
  });
});
