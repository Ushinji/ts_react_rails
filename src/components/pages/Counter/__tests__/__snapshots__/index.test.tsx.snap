// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Counter Snapshot test 1`] = `
<DocumentFragment>
  .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 18px;
  border-radius: 8px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #3173dc;
  color: #ffffff;
  padding: 8px 16px;
  width: 100%;
  -webkit-transition: 0.2s;
  transition: 0.2s;
  outline: none;
}

.c2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ffffff;
  background-color: #266cda;
  cursor: pointer;
}

.c2:hover:disabled {
  cursor: not-allowed;
}

.c2:disabled,
.c2:disabled:hover {
  opacity: 0.5;
  cursor: not-allowed;
}

.c0 {
  background-color: white;
  box-shadow: 0px 0px 4px rgba(51,23,20,0.2);
  padding: 16px;
  width: 340px;
  border-radius: 4px;
}

.c1 {
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 16px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      カウンター：0
    </div>
    <button
      class="c2"
      type="button"
    >
      Count up
    </button>
  </div>
</DocumentFragment>
`;
